Ardupilog.m

```diff
@@ -308,6 +314,53 @@ function delete(obj)
             return;
         end
 
+        function slice = getSlice(obj, slice_values, slice_type)
+        % This returns an indexed portion (a "slice") of an Ardupilog
+        % Example:
+        %    log_during_cruise = Log.getSlice([t_begin_cruise, t_end_cruise], 'TimeUS')
+        %  will return a smaller Ardupilog, only containing log data between
+        %  TimeUS values greater than t_begin_cruise and less than t_end_cruise.
+
+            % Copy all the properties, zero the number of messages
+            slice = copy(obj);
+            slice.numMsgs = 0;
+            
+            % Loop through the LogMsgGroups, slicing each one
+            for msgGroup = slice.logMsgGroups
```

Comment:
For copying, I think we need some way to loop through all the LogMsgGroups in an Ardupilog and copy each.

One way to do it is keep the meta.DynamicProperty objects created during the "addprop()" call.

Another way I can think of, which I think you utilized in #50 is to loop through all properties and query each with isa(property, 'LogMsgGroup') before proceeding.

Do you prefer the 2nd way, or perhaps another way I haven't written here? I don't have a preference.